File: backend/app/schemas/rekordbox.py
from pydantic import BaseModel


class RekordboxExportResult(BaseModel):
    tracks_added: int
    tracks_skipped: int
    playlist_updated: int
    xml_path: str
    playlist_name: str
    is_rekordbox_running: bool


class RekordboxStatus(BaseModel):
    platform: str
    xml_exists: bool
    xml_path: str
    total_tracks: int
    total_playlists: int
    detected_paths: list[str] = []

File: frontend/src/api/rekordbox.ts
import { api } from "./client";

export interface RekordboxExportResult {
  tracks_added: number;
  tracks_skipped: number;
  playlist_updated: number;
  xml_path: string;
  playlist_name: string;
  is_rekordbox_running: boolean;
}

export interface RekordboxStatus {
  platform: string;
  xml_exists: boolean;
  xml_path: string;
  total_tracks: number;
  total_playlists: number;
  detected_paths: string[];
}

export const rekordboxApi = {
  export: (sourceId: number) =>
    api.post<RekordboxExportResult>(`/rekordbox/${sourceId}/export`),
  status: () => api.get<RekordboxStatus>("/rekordbox/status"),
  discover: () => api.get<{ detected_paths: string[] }>("/rekordbox/discover"),
};
